app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/PuzzleDirectoryUI.kt

```diff
@@ -0,0 +1,418 @@
+                                                            Column(modifier = Modifier.padding(8.dp)) {
+                                                                Text(
+                                                                    text = it.name,
+                                                                    style = TextStyle.Default.copy(
+                                                                        fontSize = 12.sp,
+                                                                        fontWeight = FontWeight.Bold
+                                                                    )
+                                                                )
+                                                                it.owner?.let {
+                                                                    val flag = convertCountryCodeToEmojiFlag(it.country)
+                                                                    val ago = getRelativeTimeSpanString((ts ?: now()).toEpochMilli())
+                                                                    Text(
+                                                                        text = "by ${it.username} $flag - visited $ago",
+                                                                        style = TextStyle.Default.copy(
+                                                                            fontSize = 8.sp,
+                                                                            fontWeight = FontWeight.Light
+                                                                        )
+                                                                    )
+                                                                }
+                                                            }
+                                                        }
+                                                    }
+                                                }
+                                            } ?: Surface(
+                                                shape = MaterialTheme.shapes.medium,
+                                                modifier = Modifier.height(50.dp)
+                                            ) {}
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        resultCollections.ifEmpty { null }?.let { collections ->
+            items(items = collections) {
+                Surface(
+                    shape = MaterialTheme.shapes.medium,
+                    modifier = Modifier
+                        .height(150.dp)
+                        .fillMaxWidth()
+                        .padding(horizontal = 8.dp, vertical = 4.dp)
+                ) {
+                    Row(modifier = Modifier
```

Comment:
There is too much info and text displayed for a mobile app. Please consider removing most of the stuff that can be removed and still keep functionality (such as the number of views or the number of solves). The whole idea of the app is to provide a streamlined experience to the user, if the user wants all the features he or she can just use the clunky website.

Reply:
I'd rather not drive people to using the website if possible, that's what most of my efforts have been in the interest of. I don't mind hiding some of this though, if it's still accessible, somewhere, I just don't all the ideas for that, maybe a popup for further detail? More little flags like the one for "visited"?